### PR TITLE
AWS: Add kubernetes-e2e-aws job to hack/jenkins/e2e.sh

### PR DESCRIPTION
**Pull request by justinsb**
We'll probably have to do something more scalable than duplicate every
GCE job, but this is probably the most important job to have.

### CODE REVIEW

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -74,6 +74,11 @@ if [[ ${JOB_NAME} =~ ^kubernetes-.*-gce ]]; then
 elif [[ ${JOB_NAME} =~ ^kubernetes-.*-gke ]]; then
   KUBERNETES_PROVIDER="gke"
   : ${E2E_ZONE:="us-central1-f"}
+elif [[ ${JOB_NAME} =~ ^kubernetes-.*-aws ]]; then
+  KUBERNETES_PROVIDER="aws"
+  : ${E2E_MIN_STARTUP_PODS:="1"}
+#  : ${E2E_ZONE:="us-central1-f"}
```
Comment (ashcrow) [inline hack/jenkins/e2e.sh]: I think this line slipped in. If the user will need to set up the ```E2E_ZONE``` but uncommenting and filling in the field I recommend either setting a sane default which can be overridden or erroring out noting it's required.

**Comment by justinsb** [comment]
You're absolutely right @ashcrow - thanks (the syntax was new to me; I left it as a todo, but forgot to come back to it).  I set the default to us-east-1a.

**Comment by ashcrow** [comment]
@justinsb No issue al all, that's what reviews are for :smiley: 

I'm not able to test this myself but this looks correct to me.

**Comment by ashcrow** [comment]
I don't think @k8s-teamcity-mesosphere's build failure is related.

**Comment by justinsb** [comment]
Rebased to force a retest, to make sure that mesophere fail was a flake

**Comment by justinsb** [comment]
@eparis are you the right person to review this one?

**Comment by justinsb** [comment]
ping @eparis ... this is blocking running e2e tests on AWS.

**Pull request merged**